index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+                      last_updated: 1601979232403
+                      profile_hash: "c24d14c2c75f55d334a7e0ccf4d35a063a2582a7abb91e16d326f6613b9602bf"
+                posted:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "posted"
+                      last_updated: 1601979232403
+                      profile_hash: "c24d14c2c75f55d334a7e0ccf4d35a063a2582a7abb91e16d326f6613b9602bf"
+                deleted:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "deleted"
+                      last_updated: 1601979232403
+                      profile_hash: "c24d14c2c75f55d334a7e0ccf4d35a063a2582a7abb91e16d326f6613b9602bf"
+                validation_failed_no_profile_url:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "validation_failed"
+                      failure_reasons:
+                        - "Could not read from profile_url: https://somenode.org/optional-subdirectory/node-profile.json"
+                validation_failed_with_schema:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "validation_failed"
+                      failure_reasons:
+                        - "example_schema-v1.geolocation.lat: Invalid type. Expected: number, given: string"
+                validation_failed_no_schema:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "validation_failed"
+                      failure_reasons:
+                        - "Could not read from schema: example_schema-v1"
+                post_failed:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "post_failed"
+                      last_updated: 1601979232403
+                NodeNotFoundInIndex:
```

Comment:
Let's talk about how we use response codes for errors at different layers in the API. Mentioning here because I think this would be a legit place to use a 404 (since a specific resource was requested and not found). But, I'm happy that it's not using a 404, because I'm in favor of using HTTP error responses only for situations where the request to the index was fundamentally broken (ie, non-existing endpoint, missing required parameters for the endpoint, rate limits, authentication, etc.) I think we could think this through a bit, and see what sort of consistent pattern we could settle on.

Reply:
OK. I flagged it as a discussion point when we speak next.